examples/nitro.c

```diff
@@ -124,6 +154,48 @@ int main(int argc, char *const argv[])
 
     }
 
+    // Create and initialize UNIX IPC socket for reading enclave output.
+    sock_fd = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (sock_fd < 0) {
+        perror("Error creating UNIX IPC socket for enclave communication");
+        return -1;
+    }
+    memset(&addr, 0, sizeof(struct sockaddr_un));
+    addr.sun_family = AF_UNIX;
+    strcpy(addr.sun_path, IPC_SOCK_PATH);
+
+    // Listen on the socket for enclave output.
+    unlink(IPC_SOCK_PATH);
+    ret = bind(sock_fd, (struct sockaddr *) &addr, sizeof(addr));
+    if (ret < 0) {
+        perror("Error binding socket");
+        close(sock_fd);
+        exit(1);
+    }
+
+    ret = listen(sock_fd, 1);
+    if (ret < 0) {
+        perror("Error listening on socket");
+        close(sock_fd);
+        exit(1);
+    }
+
+    // Configure the IPC socket to read output from the enclave. The "port"
+    // argument is ignored.
+    if (err = krun_add_vsock_port(ctx_id, 0, IPC_SOCK_PATH)) {
+        errno = -err;
+        perror("Error configuring enclave vsock");
```

Comment:
Would we be leaking the socket fd here and in `krun_start_enter`?

Reply:
What do you mean by "leaking"?

Reply:
On line 165 you create a socket, but if `krun_add_vsock_port`  or `krun_start_enter` fails the socket never gets closed via `close(sock_fd)` and the resources are "leaked".

Reply:
It should be cleaned up by the OS iirc but I think it's best practice to clean them up manually regardless.

Reply:
I suppose it would be good practice to close the socket fd, but it will be closed anyway when the process exits.

Reply:
> It should be cleaned up by the OS iirc but I think it's best practice to clean them up manually regardless.

Sure, will do so.